commons/src/main/java/org/infinispan/commons/util/InfinispanCollections.java

```diff
@@ -301,15 +302,29 @@ public int size() {
     * @see #EMPTY_LIST
     */
    @SuppressWarnings("unchecked")
-   public static final <T> List<T> emptyList() {
+   public static <T> List<T> emptyList() {
       return EMPTY_LIST;
    }
 
-   public static final <T> boolean containsAny(Collection<T> haystack, Collection<T> needle) {
+   public static <T> boolean containsAny(Collection<T> haystack, Collection<T> needle) {
       for (T element : needle) {
          if (haystack.contains(element))
             return true;
       }
       return false;
    }
+
+   public static <T> void forEach(T[] array, Consumer<T> consumer) {
+      Objects.requireNonNull(array, "Array must be non-null.");
+      final int size = array.length;
+      if (size == 0) {
+         return;
+      } else if (size == 1) {
+         consumer.accept(array[0]);
+         return;
+      }
+      for (int i = 0; i < size; ++i) {
```

Comment:
I think only the for loop should be good enough - there's no allocation, and it will do the same `0 < size`, `1 < size` comparisons.